AK/FixedArray.h

```diff
@@ -15,40 +17,61 @@ namespace AK {
 template<typename T>
 class FixedArray {
 public:
-    FixedArray() = default;
-    explicit FixedArray(size_t size)
-        : m_size(size)
+    static FixedArray<T> empty()
```

Comment:
What's the point of this? Just keep the default constructor.

Reply:
The point is to not mix public construction functions and constructors. I think it's far cleaner if all constructors are either public or private.

Reply:
That's neither wrong nor unusual. Look at `Core::DateTime` or `Keyboard::CharacterMap`, for example.

Reply:
The public `Core::DateTime` constructor is automatically generated, right?

Reply:
Okay, I guess you're right. I still don't feel great about mixing these, but if you think the resulting code is clearer, I will restore the public default constructor.

Reply:
> The public Core::DateTime constructor is automatically generated, right?

Right :)